### PR TITLE
TDL-731: Added discovery test from tap-tester to test folder.

### PR DESCRIPTION
**Pull request by dbshah1212**
# Description of change
TDL-731: Tests have been moved out of tap-tester and into tap-pipedrive and changed those to work in tap-pipedrive and updated config.yml for the same.

# Manual QA steps
 -Not required
 
# Risks
 No Risk
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: 2.1
 jobs:
   build:
     docker:
-      - image: 218546966473.dkr.ecr.us-east-1.amazonaws.com/circle-ci:tap-tester
+      - image: 218546966473.dkr.ecr.us-east-1.amazonaws.com/tap-tester:4.latest
```
Comment (karanpanchal-crest) [inline .circleci/config.yml:5]: @dbshah1212 Is it intentional for changing this version?

Comment (kpanchal-crest) [reply .circleci/config.yml:5]: Offline discussed with Dhruvin.. It is intentional

Comment (kpanchal-crest) [reply .circleci/config.yml:5]: circle-ci:tap-tester-v4 -> Please change the version

Comment (karanpanchal-crest) [reply .circleci/config.yml:5]: circle-ci:tap-tester-v4 -> Please change the version

**Comment by dbshah1212** [comment]
delete_logs tests will be removed as part of the delete_log card.

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,19 +12,26 @@ jobs:
...
+      - run:
+          name: 'Unit Tests'
+          command: |
+            source /usr/local/share/virtualenvs/tap-pipedrive/bin/activate
+            python -m unittest discover tests/unittests
+      - add_ssh_keys
       - run:
           name: 'Integration Tests'
           command: |
             aws s3 cp s3://com-stitchdata-dev-deployment-assets/environments/tap-tester/sandbox dev_env.sh
```
Comment (kspeer825) [inline .circleci/config.yml:24]: This should be tap_tester_sandbox not just sandbox.

Comment (dbshah1212) [reply .circleci/config.yml]: @kspeer825 Updated.

File: tests/base.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+        return {table: properties.get(self.FOREIGN_KEYS, set())
+                for table, properties
+                in self.expected_metadata().items()}
+
+
+    def expected_automatic_fields(self):
+        auto_fields = {}
+        for k, v in self.expected_metadata().items():
+            auto_fields[k] = v.get(self.PRIMARY_KEYS, set()) | v.get(self.REPLICATION_KEYS, set()) \
+                | v.get(self.FOREIGN_KEYS, set())
```
Comment (kspeer825) [inline tests/base.py:189]: Foreign Keys are not expected to be automatic. This is my fault, bad copy pasta.

Comment (kspeer825) [reply tests/base.py]: Fixed.

**Review by kspeer825 — APPROVED** [review]

**Review by KAllan357 — APPROVED** [review]

**Pull request merged**